src/nodes/hypertable_modify.c

```diff
@@ -93,6 +110,286 @@ get_chunk_dispatch_states(PlanState *substate)
+	tupleDesc = RelationGetDescr(chunk_rel);
+	*attrno = get_attnum(curr_chunk->table_id, column_name);
+	if (!AttributeNumberIsValid(*attrno))
+		elog(ERROR,
+			 "Could not find column: \"%s\" in table: \"%s\"",
+			 column_name,
+			 get_rel_name(curr_chunk->table_id));
+	/*
+	 * check if SEGMENT BY column is NULLABLE or not based on uncompressed
+	 * chunk tuple descriptor. This is because compressed chunk column
+	 * attributes don't have NULLABLE properties.
+	 */
+	Form_pg_attribute attribute = TupleDescAttr(tupleDesc, AttrNumberGetAttrOffset(*attrno));
+	is_null = attribute->attnotnull;
+	table_close(chunk_rel, AccessShareLock);
+	/*
+	 * we get attrno for SEGMENT BY column from uncompressed chunk. This
+	 * can be different from that of compressed chunk due to different
+	 * physical layouts of compressed and uncompressed chunks. Thus we
+	 * need to retreive correct attrno from compressed chunk tuple descriptor.
+	 */
+	comp_chunk_rel = table_open(curr_comp_chunk->table_id, AccessShareLock);
+	tupleDesc = RelationGetDescr(comp_chunk_rel);
+
+	*attrno = get_attnum(curr_comp_chunk->table_id, column_name);
+	if (!AttributeNumberIsValid(*attrno))
+		elog(ERROR,
+			 "Could not find column: \"%s\" in table: \"%s\"",
+			 column_name,
+			 get_rel_name(curr_chunk->table_id));
+	table_close(comp_chunk_rel, AccessShareLock);
+	return !is_null;
+}
+
+/*
+ * Helper method which returns true if given column_name
+ * is configured as SEGMENT BY column in a compressed hypertable
+ */
+static bool
+is_segmentby_col(List *ht_compression_info, char *column_name)
+{
+	ListCell *lc;
+	foreach (lc, ht_compression_info)
+	{
+		FormData_hypertable_compression *fd = lfirst(lc);
+		if (namestrcmp(&fd->attname, column_name) == 0)
+		{
+			if (fd->segmentby_column_index > 0)
+				return true;
+			break;
+		}
+	}
+	return false;
+}
+
+/*
+ * Helper method which will build scan keys for SEGMENT BY column
+ * with index attribute number from compressed chunk.
+ */
+static ScanKeyData *
+build_update_delete_scankeys(Chunk *curr_chunk, Chunk *curr_comp_chunk, char *column_name,
+							 Const *const_value, int strategy)
+{
+	Relation comp_chunk_rel;
+	TupleDesc tupleDesc;
+	int attno = 0;
+	Form_pg_attribute attr;
+	Datum value;
+	ScanKeyData *scankey = NULL;
+	TypeCacheEntry *tce;
+
+	comp_chunk_rel = table_open(curr_comp_chunk->table_id, AccessShareLock);
+	tupleDesc = RelationGetDescr(comp_chunk_rel);
+	scankey = palloc0(sizeof(ScanKeyData));
+	value = (const_value ? const_value->constvalue : 0);
+	/*
+	 * make compiler happy by silencing this error
+	 * ‘attr’ may be used uninitialized in this function
+	 */
+	attr = TupleDescAttr(tupleDesc, 0);
+	for (attno = 1; attno <= tupleDesc->natts; attno++)
+	{
+		attr = TupleDescAttr(tupleDesc, AttrNumberGetAttrOffset(attno));
+		if (namestrcmp(&(attr->attname), column_name) == 0)
+			break;
+	}
+	tce = lookup_type_cache(attr->atttypid, TYPECACHE_EQ_OPR_FINFO);
```

Comment:
While lookup_type_cache will always return something, the specific information requested might still be InvalidOid so we nned to check for that

Reply:
Ok, Added this additional check:
```
	if (NULL == tce || !OidIsValid(tce->eq_opr_finfo.fn_oid))
		elog(ERROR, "no equality function for column \"%s\"", column_name);
```

Reply:
lookup_type_cache will never return NULL